assets/js/modules/analytics-4/datastore/conversion-reporting.js

```diff
@@ -16,42 +16,34 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
 /**
  * Internal dependencies
  */
 import API from 'googlesitekit-api';
-import { combineStores, createRegistrySelector } from 'googlesitekit-data';
+import {
+	commonActions,
+	combineStores,
+	createRegistrySelector,
+	createReducer,
+} from 'googlesitekit-data';
 import { MODULES_ANALYTICS_4 } from './constants';
 import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
 
-export const selectors = {
-	/**
-	 * Checks whether the provided conversion reporting events are available.
-	 *
-	 * @since 1.135.0
-	 *
-	 * @param {Object}               state  Data store's state.
-	 * @param {string|Array<string>} events Conversion reporting events to check.
-	 * @return {(boolean|undefined)} True if all provided custom dimensions are available, otherwise false. Undefined if available custom dimensions are not loaded yet.
-	 */
-	hasConversionReportingEvents: createRegistrySelector(
-		( select ) => ( state, events ) => {
-			// Ensure events is always an array, even if a string is passed.
-			const eventsToCheck = Array.isArray( events ) ? events : [ events ];
-
-			const detectedEvents =
-				select( MODULES_ANALYTICS_4 ).getDetectedEvents();
+function hasConversionReportingEventsOfType( propName ) {
+	return createRegistrySelector( ( select ) => () => {
+		const inlineData =
+			select(
+				MODULES_ANALYTICS_4
+			).getConversionReportingEventsChange() || {};
 
-			if ( ! detectedEvents?.length ) {
-				return false;
-			}
-
-			return eventsToCheck.some( ( event ) =>
-				detectedEvents.includes( event )
-			);
-		}
-	),
-};
+		return inlineData[ propName ]?.length;
```

Comment:
This should return a boolean but it will return `int|undefined`. We want to keep the undefined return but change the int to boolean. This can be addressed using the `negateDefined` utility. You can see an example in `isKeyMetricsSetupCompleted`